stl/inc/xutility

```diff
@@ -2101,16 +2108,12 @@ namespace ranges {
 
             template <class _Ty>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
-                constexpr bool _Is_lvalue = is_lvalue_reference_v<_Ty>;
-
                 if constexpr (is_array_v<remove_reference_t<_Ty>>) {
-                    if constexpr (_Is_lvalue) {
-                        return {_St::_Array, true};
-                    }
-                } else if constexpr (_Is_lvalue && _Has_member<_Ty>) {
-                    return {_St::_Member, noexcept(_Fake_decay_copy(_STD declval<_Ty>().begin()))};
-                } else if constexpr (_Has_ADL<_Ty>) {
-                    return {_St::_Non_member, noexcept(_Fake_decay_copy(begin(_STD declval<_Ty>())))};
+                    return {_St::_Array, true};
+                } else if constexpr (_Has_member<_Ty&>) {
+                    return {_St::_Member, noexcept(_Fake_decay_copy(_STD declval<_Ty&>().begin()))};
+                } else if constexpr (_Has_ADL<_Ty&>) {
+                    return {_St::_Non_member, noexcept(_Fake_decay_copy(begin(_STD declval<_Ty&>())))};
                 }

```

Comment:
Previously, the final `return` handled multiple cases. Now, there are 4 disjoint cases. I believe that this should be an `else` so that exactly one `return` is selected. (For runtime `if` statements, we prefer "early `return`" to reduce indentation, but for `if constexpr`, we generally prefer the form that compiles the least code, which means using `else` in situations like this.)